debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/TypeRegistry.java

```diff
@@ -39,6 +51,22 @@
     public static final int NO_TYPE_MODIFIER = -1;
     public static final int UNKNOWN_LENGTH = -1;
 
+    private static final String SQL_NON_ARRAY_TYPES = "SELECT t.oid AS oid, t.typname AS name, t.typbasetype AS baseoid, t.typtypmod as modifiers "
```

Comment:
Have you tried to analyze the stateents to make sure they will not take too long time in case of very large databases - e.g. many types etc.?

Reply:
I haven't but I will simulate it and let you know.

Reply:
@jpechane I wanted to circle back to this because this particular query is nearly unchanged from its original version in `PostgresConnection`.  The only main differences from the original query are the following:

* Added t.typbasetype and t.typtypmod columns to select clause
* Applied `ORDER BY t.typbasetype`

That said, based on your comment below about `resolveUnknownType`, we'll need to change this up slightly at least to introduce a predicate to only pickup non-domain types here and remove the order-by requirement; which should bring it back moreso to its original version.